src/libasr/pass/intrinsic_functions.h

```diff
@@ -1066,6 +1067,57 @@ namespace Shiftl {
 
 } // namespace Shiftl
 
+namespace Dshiftl {
+
+        static ASR::expr_t *eval_Dshiftl(Allocator &al, const Location &loc,
+                ASR::ttype_t* t1, Vec<ASR::expr_t*> &args, diag::Diagnostics& diag) {
+        int64_t val1 = ASR::down_cast<ASR::IntegerConstant_t>(args[0])->m_n;
+        int64_t val2 = ASR::down_cast<ASR::IntegerConstant_t>(args[1])->m_n;
+        int64_t shift = ASR::down_cast<ASR::IntegerConstant_t>(args[2])->m_n;
+        int kind1 = ASRUtils::extract_kind_from_ttype_t(ASR::down_cast<ASR::IntegerConstant_t>(args[0])->m_type);
+        int kind2 = ASRUtils::extract_kind_from_ttype_t(ASR::down_cast<ASR::IntegerConstant_t>(args[1])->m_type);
+        if(kind1 != kind2) {
+            append_error(diag, "The kind of first argument of 'dshiftl' intrinsic must be the same as second arguement", loc);
+            return nullptr;
+        }
+        if(shift < 0){
+            append_error(diag, "The shift argument of 'dshiftl' intrinsic must be non-negative integer", loc);
+            return nullptr;
+        }
+        if(kind1 == 8){
+            int64_t val = (val1 << shift) | (val2 >> (64 - shift));
+            return make_ConstantWithType(make_IntegerConstant_t, val, t1, loc);
+        }
+        int64_t val = (val1 << shift) | (val2 >> (32 - shift));
+        return make_ConstantWithType(make_IntegerConstant_t, val, t1, loc);
+    }
+
+
+    static inline ASR::expr_t* instantiate_Dshiftl(Allocator &al, const Location &loc,
+            SymbolTable *scope, Vec<ASR::ttype_t*>& arg_types, ASR::ttype_t *return_type,
+            Vec<ASR::call_arg_t>& new_args, int64_t /*overload_id*/) {
+        declare_basic_variables("_lcompilers_dshiftl_" + type_to_str_python(arg_types[0]));
+        fill_func_arg("i", arg_types[0]);
+        fill_func_arg("j", arg_types[1]);
+        fill_func_arg("shift", arg_types[2]);
+        auto result = declare(fn_name, return_type, ReturnVar);
+        /*
+        * r = Dshiftl(x, y, shift)
+        * r = x << shift | y >> (32 - shift)
```

Comment:
Also, let's have the same implementation in both compile time and run time
```suggestion
        * r = x << shift | y >> (32 - shift) ! kind = 4
        or
        * r = x << shift | y >> (64 - shift) ! kind = 8
```